timemachine/fe/estimator.py

```diff
@@ -184,35 +262,115 @@ def deltaG(model, sys_params) -> Tuple[float, List]:
         bp = unbound_pot.bind(np.asarray(params))
         bound_potentials.append(bp)
 
-    client = model.client
-    if client is None:
-        client = SerialClient()
-        client.verify()
+    if model.endpoint_correct:
+        sim_results = results[:-1]
+    else:
+        sim_results = results
 
-    futures = []
-    for lamb in model.lambda_schedule:
-        args = (
-            lamb,
-            model.box,
-            model.x0,
-            model.v0,
-            bound_potentials,
-            model.integrator,
-            model.equil_steps,
-            model.prod_steps,
-            model.barostat,
+    U_knk = []
+    N_k = []
+    for result in sim_results:
+        U_knk.append(result.lambda_us)
+        N_k.append(len(result.lambda_us))  # number of frames
+
+    U_knk = np.array(U_knk)
+
+    bar_dG = 0
+    bar_dG_err = 0
+
+    delta_Us = extract_delta_Us_from_U_knk(U_knk)
+
+    for lambda_idx in range(len(model.lambda_schedule) - 1):
```

Comment:
nit: style preference for
```suggestion
    for lambda_idx, (lamb_start, lamb_end) in enumerate(zip(model.lambda_schedule, model.lambda_schedule[1:])):
```